pkg/scheduler/plugins/overcommit/overcommit.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2021 The Volcano Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package overcommit
+
+import (
+	"k8s.io/klog"
+	"time"
+	"volcano.sh/volcano/pkg/apis/scheduling"
+	"volcano.sh/volcano/pkg/scheduler/api"
+	"volcano.sh/volcano/pkg/scheduler/framework"
+)
+
+const (
+	// PluginName is name of plugin
+	PluginName = "overcommit"
+	// overCommitFactor is resource overCommit factor for enqueue action
+	// It determines the number of `pending` pods that the scheduler will tolerate
+	// when the resources of the cluster is insufficient
+	overCommitFactor = "overcommit-factor"
+	// defaultOverCommitFactor defines the default overCommit resource factor for enqueue action
+	defaultOverCommitFactor = 1.2
+)
+
+type overcommitPlugin struct {
+	// Arguments given for the plugin
+	pluginArguments  framework.Arguments
+	idleResource     *api.Resource
+	inqueueResource  *api.Resource
+	overCommitFactor float64
+}
+
+// New function returns overcommit plugin object
+func New(arguments framework.Arguments) framework.Plugin {
+	return &overcommitPlugin{
+		pluginArguments:  arguments,
+		idleResource:     api.EmptyResource(),
+		overCommitFactor: defaultOverCommitFactor,
+	}
+}
+
+func (op *overcommitPlugin) Name() string {
+	return PluginName
+}
+
+func (op *overcommitPlugin) OnSessionOpen(ssn *framework.Session) {
+	overcommitStartTime := time.Now().UnixNano()
+	op.pluginArguments.GetFloat64(&op.overCommitFactor, overCommitFactor)
```

Comment:
add overCommitFactor illegal check

Reply:
Added overcommit-factor validity check: if input < 1, print warning logs & use default value instead.

Reply:
what happen if `overcommit-factor` is an invalid number?

Reply:
> what happen if `overcommit-factor` is an invalid number?

`GetFloat64()` do not return err, only print warning logs, so here initializes `overcommit-factor` with an illegal value `-1`. If its value is not updated after `GetFloat64()`, it will be processed the sam as condition 'input < 1'.
https://github.com/volcano-sh/volcano/blob/c0fe287d6299082bc407ebf2c7bede0db9c78386/pkg/scheduler/framework/arguments.go#L51